expose and fix bug in estimate_min_max_CoverFreq for non-fixed sky locations

dbkeitel opened the pull request:
First commit splits the MCMCSearch test case into two: 
 - test_fully_coherent_MCMC_fixedsky
 - test_fully_coherent_MCMC_skyprior
 - the latter exposes bug from https://github.com/PyFstat/PyFstat/issues/106

@Rodrigo-Tenorio will look into a fix

Rodrigo-Tenorio commented:
If I'm not mistaken, it looks like the problem was a step defined as 0 when it should have really been `upper - lower`.

dbkeitel commented:
Thanks, you've found the source of the problem, but I think the right solution is to just drop that third entry in the list - `ComputeFstat.estimate_min_max_CoverFreq()` will accept `either [single_value], [min,max] or [min,max,step]` and so you're now basically telling it the MCMC has a "step size" of the whole covered band. But anyway the step size is not needed for the coverage band estimation, so we can just fall back on the default.

I'll push a new version.

dbkeitel commented on pyfstat/mcmc_based_searches.py:
```diff
@@ -235,8 +235,8 @@ def _get_search_ranges(self):
                 )
             # first start with parameters that have non-delta prior ranges
             search_ranges = {
-                key: [prior_bounds[key]["lower"], prior_bounds[key]["upper"], 0]
```
Thanks, you've found the source of the problem, but I think the right solution is to just drop that third entry in the list - `ComputeFstat.estimate_min_max_CoverFreq()` will accept `either [single_value], [min,max] or [min,max,step]` and so you're now basically telling it the MCMC has a "step size" of the whole covered band. But anyway the step size is not needed for the coverage band estimation, so we can just fall back on the default.

I'll push a new version.

dbkeitel commented on pyfstat/mcmc_based_searches.py:
```diff
@@ -235,8 +235,8 @@ def _get_search_ranges(self):
-                key: [prior_bounds[key]["lower"], prior_bounds[key]["upper"], 0]
-                for key in prior_bounds.keys()
+                key: [bound["lower"], bound["upper"], bound["upper"] - bound["lower"]]
+                for key, bound in prior_bounds.items()
```
Thanks, you've found the source of the problem, but I think the right solution is to just drop that third entry in the list - `ComputeFstat.estimate_min_max_CoverFreq()` will accept `either [single_value], [min,max] or [min,max,step]` and so you're now basically telling it the MCMC has a "step size" of the whole covered band. But anyway the step size is not needed for the coverage band estimation, so we can just fall back on the default.

I'll push a new version.

Rodrigo-Tenorio replied:
That actually makes sense.

dbkeitel commented:
I've squashed my version into your commit (but as I had hoped in adding an extra in-line comment, the previous suggested solution is still preserved as a "resolved conversation" above).

Rerunning the test case a few times, I also realised Gaussian priors on the sky are a bad idea, since the walkers can jump out of the allowed delta range. Changed to uniform.

Rodrigo-Tenorio commented on tests.py:
```diff
@@ -697,54 +697,113 @@ def test_get_semicoherent_nglitch_twoF(self):
-            "Alpha": Alpha,
-            "Delta": Delta,
+            "F0": {"type": "norm", "loc": self.F0, "scale": np.abs(1e-10 * self.F0)},
+            "F1": {"type": "norm", "loc": self.F1, "scale": np.abs(1e-10 * self.F1)},
```
I thought you said uniform? Or do you do this in order to test the normalization warning as well?

dbkeitel replied:
I changed Alpha,Delta to uniform in the "skyprior" case below, but left F0,F1 alone in both.

Rodrigo-Tenorio commented on tests.py:
```diff
@@ -697,54 +697,113 @@ def test_get_semicoherent_nglitch_twoF(self):
+            "F0": {"type": "norm", "loc": self.F0, "scale": np.abs(1e-10 * self.F0)},
+            "F1": {"type": "norm", "loc": self.F1, "scale": np.abs(1e-10 * self.F1)},
+            "F2": self.F2,
+            "Alpha": {
```
This may put us in some trouble if Alpha or Delta are close to their edges (it already happened to me); however, I'm willing to believe 0.01 is small enough so this never happens _almost surely_.

dbkeitel replied:
Yes doing it that way could cause trouble in general, but the test suite is set up with globally
```
        self.Alpha = 5e-3
        self.Delta = 1.2
```
so it's safe here.
